collector/teradici_pcoip.go

```diff
@@ -0,0 +1,665 @@
+//go:build windows
+// +build windows
+
+package collector
+
+import (
+	"errors"
+
+	"github.com/StackExchange/wmi"
+	"github.com/prometheus-community/windows_exporter/log"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+func init() {
+	registerCollector("teradici_pcoip", NewTeradiciPcoipCollector)
+}
+
+// A TeradiciPcoipCollector is a Prometheus collector for WMI metrics:
+// Win32_PerfRawData_TeradiciPerf_PCoIPSessionAudioStatistics
+// Win32_PerfRawData_TeradiciPerf_PCoIPSessionGeneralStatistics
+// Win32_PerfRawData_TeradiciPerf_PCoIPSessionImagingStatistics
+// Win32_PerfRawData_TeradiciPerf_PCoIPSessionNetworkStatistics
+// Win32_PerfRawData_TeradiciPerf_PCoIPSessionUsbStatistics
+
+type TeradiciPcoipCollector struct {
+	AudioBytesReceived       *prometheus.Desc
+	AudioBytesSent           *prometheus.Desc
+	AudioRXBWkbitPersec      *prometheus.Desc
+	AudioTXBWkbitPersec      *prometheus.Desc
+	AudioTXBWLimitkbitPersec *prometheus.Desc
+
+	BytesReceived          *prometheus.Desc
+	BytesSent              *prometheus.Desc
+	PacketsReceived        *prometheus.Desc
+	PacketsSent            *prometheus.Desc
+	RXPacketsLost          *prometheus.Desc
+	SessionDurationSeconds *prometheus.Desc
+	TXPacketsLost          *prometheus.Desc
+
+	ImagingActiveMinimumQuality        *prometheus.Desc
+	ImagingApex2800Offload             *prometheus.Desc
+	ImagingBytesReceived               *prometheus.Desc
+	ImagingBytesSent                   *prometheus.Desc
+	ImagingDecoderCapabilitykbitPersec *prometheus.Desc
+	ImagingEncodedFramesPersec         *prometheus.Desc
+	ImagingMegapixelPersec             *prometheus.Desc
+	ImagingNegativeAcknowledgements    *prometheus.Desc
+	ImagingRXBWkbitPersec              *prometheus.Desc
+	ImagingSVGAdevTapframesPersec      *prometheus.Desc
+	ImagingTXBWkbitPersec              *prometheus.Desc
+
+	RoundTripLatencyms        *prometheus.Desc
+	RXBWkbitPersec            *prometheus.Desc
+	RXBWPeakkbitPersec        *prometheus.Desc
+	RXPacketLossPercent       *prometheus.Desc
+	RXPacketLossPercent_Base  *prometheus.Desc
+	TXBWActiveLimitkbitPersec *prometheus.Desc
+	TXBWkbitPersec            *prometheus.Desc
+	TXBWLimitkbitPersec       *prometheus.Desc
+	TXPacketLossPercent       *prometheus.Desc
+	TXPacketLossPercent_Base  *prometheus.Desc
+
+	USBBytesReceived  *prometheus.Desc
+	USBBytesSent      *prometheus.Desc
+	USBRXBWkbitPersec *prometheus.Desc
+	USBTXBWkbitPersec *prometheus.Desc
+}
+
+// NewTeradiciPcoipCollector constructs a new TeradiciPcoipCollector
+func NewTeradiciPcoipCollector() (Collector, error) {
+	const subsystem = "teradici_pcoip"
+	return &TeradiciPcoipCollector{
+		AudioBytesReceived: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "audio_bytes_received"),
```

Comment:
All metrics of type `counter` will need to be suffixed with "_total". In this case, `audio_bytes_received` -> `audio_bytes_received_total`.

Reply:
Thanks for catching this - should now be resolved